### DISCUSSION TITLE
April Developer Meeting

### DISCUSSION THREAD

**Discussion by TheDuckCow** [Moo-Ack-Productions/MCprep#560] [General Discussions]
Discussion for the latest MCprep developer meeting, starting now ~~[to be edited later]~~ Notes below!

**Comment by TheDuckCow** [comment]
Notes:

- common obj spec:
  - Do we want to save the block raw str for a spec (to support better forward compatibility);
  - Right now, all properties of the mc common is stored to each _object_; we could check the version and do re-parsing at the time of relevant
  - Could add an operator to "update common obj refs" to go through everything. Though auto udpating as needed would be nice, the spec is needed for some UI drawing so not always an option to do inside the opreator.
  - All the place this is relevant for existing features
    - For Meshswap (and showing meshswap options) using the exporter name
    - Texture swap
    - Animate textures (tiles vs atlas)
  - Future options:
    - Re-importing a section of a world -> for this, would 
    - Extending the world (through the fabled import bridge)
  - [See here](https://github.com/CommonMCOBJ/CommonMCOBJ#interoperability-with-selections) discussion of interoperability; we can't rely on this behavior, would mostly make the standard easier to adopt.
  - -> Takeaway: Me to review, likely to add comment to where the spec gets saved into the parent obj or the containing collection
- Bpy build: Ready for testing, should be nicely backwards compatible now. Unit tests passing locally for @StandingPadAnimations, me to validate I get the same
- I18n: Ready for QA; no actual automated tests created, me to think if there's any room for this in the meantime even if tricky.

**Comment by StandingPadAnimations** [comment]
Overall I think we're almost there to getting MCprep 3.6 released soon 

Some things to discuss that we didn't get to (since I had to leave earlier then planned):
- Updating analytics to handle CommonMCOBJ
- Getting some infastructure in place for translations (back in #540 we discussed using Weblate for translators, will need to apply for their open source project tier) 

I think once MCprep 3.6 is released, we can start brainstorming the changes for MCprep 4.0 (although its a bit further on the roadmap)

**Comment by StandingPadAnimations** [comment]
Btw @TheDuckCow, since MCprep 3.6 is a major release, does that mean we can also update our minimum supported Blender version as defined in #399 (based on the 5% rule)?
